src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Logging/RazorLogHubLogger.cs

```diff
@@ -31,18 +32,35 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
 
         switch (logLevel)
         {
+            // We separate out Information because we want to check for specific log messages set from CLaSP
+            case LogLevel.Information:
+                // The category for start and stop will only ever be "CLaSP" so no point logging it
+                if (formattedResult.StartsWith(ClaspLoggingBridge.LogStartContextMarker))
+                {
+                    _traceSource.TraceEvent(TraceEventType.Start, id: 0, "{0}", formattedResult);
+                }
+                else if (formattedResult.StartsWith(ClaspLoggingBridge.LogEndContextMarker))
+                {
+                    _traceSource.TraceEvent(TraceEventType.Stop, id: 0, "{0}", formattedResult);
+                }
```

Comment:
The above is how we further simplify what was LoggerAdapter, by not needing it have a TraceSource. Instead it logs a message in a specific format, and we catch that here to translate it into the log types that `ILogger` doesn't support. This keeps all LogHub specific logic inside the LogHub Logger, which I think is very nice.

Reply:
So this TraceSource is getting injected? Do we only have one TraceSource at all times? Or does it change every time a session is reset?

Reply:
At the moment, one.